### DISCUSSION TITLE
osu!lazer ideas

### DISCUSSION THREAD

**Discussion by zMienai** [ppy/osu#22812] [Ideas]
1. To be able to always see your grade during gameplay and update in real time
2. to be able to change the bpm of a map to whatever the player wants

**Comment by mk56-spn** [comment]
1 would be nice. And should be pretty easy to code.

There's already quite a few discussions regarding 2. I suggest you go look them up if you want more detail but TLDR just from things I can see being blockers : any BPM is unlikely to happen because it makes songs sound like crapshoot , animations get very wonky . Etc..

**Comment by zMienai** [comment]
I see where you're coming from but can't the song remain unchanged? The maps would have different bpm, but as far as i know bpm doesn't affect any of the difficulty values

**Reply by mk56-spn** [reply]
if you change the bpm you by necessity change the map speed, this can either be done through changing clock rate or moving objects closer  / farther in their timestamps, in either case a speeding up or slowing down of the track will be necessary, and if the targeted BPM is not within the current 0.5x to 2x allowed speed change it will inevitably result in more distortion to the audio than has existed up till now,

**Reply by zMienai** [reply]
Oh ok, i understand. Also could you please help me with a lag problem in osu!lazer? I posted it before this. It's called something along the lines of "EXTREME lag issues in osu!lazer"
